rinex/src/observation/mod.rs

```diff
@@ -108,69 +108,31 @@ pub struct HeaderFields {
     pub codes: HashMap<Constellation, Vec<Observable>>,
     /// True if local clock drift is compensated for
     pub clock_offset_applied: bool,
-    /// DCBs compensation per constellation basis
-    pub dcb_compensations: Vec<Constellation>,
     /// Optionnal data scalings
-    pub scalings: HashMap<Constellation, HashMap<Observable, f64>>,
+    pub scalings: HashMap<Constellation, HashMap<Observable, u16>>,
 }
 
 impl HeaderFields {
-    /// Add an optionnal data scaling
-    pub fn with_scaling(&self, c: Constellation, observable: Observable, scaling: f64) -> Self {
-        let mut s = self.clone();
-        if let Some(scalings) = s.scalings.get_mut(&c) {
+    /// Insert a data scaling
+    pub(crate) fn insert_scaling(
+        &mut self,
+        c: Constellation,
+        observable: Observable,
+        scaling: u16,
+    ) {
+        if let Some(scalings) = self.scalings.get_mut(&c) {
```

Comment:
You can use https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_default to avoid a double lookup here.

Reply:
> You can use https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_default to avoid a double lookup here (but a `HashMap<(Constellation, Observable), u16>` might be better).

The double key (Constellation, Observable) is a good idea, simplifies the overall operation. 
I'll see if I can apply that kind of simplification some place else, where we have multi dimensional arrays